sqrl-tools/sqrl-cli/src/main/java/com/datasqrl/cmd/AbstractCompilerCommand.java

```diff
@@ -84,6 +121,128 @@ public void runCommand(ErrorCollector errors) {
     }
   }
 
+  private boolean isGenerateGraphql() {
+    if (generateAPI != null) {
+      return Arrays.stream(generateAPI).anyMatch(APIType.GraphQL::equals);
+    }
+    return false;
+  }
+
+  @SneakyThrows
+  private void addGraphql(Path build, Path rootDir) {
+    Files.copy(build.resolve(GRAPHQL_NORMALIZED_FILE_NAME),
+        rootDir.resolve(GRAPHQL_NORMALIZED_FILE_NAME));
+  }
+
+  //Adds in regardless
+  private void addDockerCompose() {
+    String yml = DockerCompose.getYml();
+    Path toFile = targetDir.resolve("docker-compose.yml");
+    try {
+      Files.createDirectories(targetDir);
+      Files.writeString(toFile, yml);
+    } catch (Exception e) {
+      log.error("Could not copy docker-compose file.");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Supplier<SqrlConfig> getDefaultConfig(boolean startKafka, Path rootDir, ErrorCollector errors) {
+    if (startKafka) {
+      startKafka();
+      return ()->PackagerUtil.createEmbeddedConfig(root.rootDir, errors);
+    }
+
+    //Generate docker config
+    return ()->PackagerUtil.createDockerConfig(root.rootDir, targetDir, errors);
+  }
+
+  private void startKafka() {
+    //We're generating an embedded config, start the cluster
+    try {
+      if (!kafkaStarted) {
+        CLUSTER.start();
+      }
+      this.kafkaStarted = true;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @SneakyThrows
+  private void createGraphqlKafkaSources(Path rootDir, Path buildDir, SqrlConfig config) {
+    ResourceResolver resourceResolver = new FileResourceResolver(buildDir);
+
+    Map<String, Optional<String>> scriptFiles = ScriptConfiguration.getFiles(config);
+    Optional<URI> graphqlSchema = scriptFiles.get(ScriptConfiguration.GRAPHQL_KEY)
+            .flatMap(file -> resourceResolver.resolveFile(NamePath.of(file)));
+
+    if (graphqlSchema.isEmpty()) {
+      return;
+    }
+    Path schemaPath = Paths.get(graphqlSchema.get());
+    String schemaName = schemaPath.getFileName().toString().split("\\.")[0];
+
+    String schema = Files.readString(schemaPath);
+    TypeDefinitionRegistry registry = new SchemaParser().parse(schema);
+
+    ObjectTypeDefinition mutationType = (ObjectTypeDefinition) registry
+            .getType("Mutation")
+            .orElse(null);
+    if (mutationType == null) {
+      return;
+    }
+
+    Path schemaRoot = buildDir.resolve(schemaName);
+    Files.createDirectories(schemaRoot);
+    for (graphql.language.FieldDefinition definition : mutationType.getFieldDefinitions()) {
+      String bootstrapServers;
+      if (kafkaStarted) {
+        CLUSTER.createTopic(definition.getName());
+        bootstrapServers = CLUSTER.bootstrapServers();
+      } else {
+        bootstrapServers = "kafka:9092";
```

Comment:
To use mutations the user has to use docker?

Reply:
As we talked about, kafka as an engine type is not yet implemented. They are handled as sources and sinks. If you read the ticket, you'll see the full scope of work.